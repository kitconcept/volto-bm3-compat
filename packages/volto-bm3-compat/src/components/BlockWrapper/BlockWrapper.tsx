import MaybeWrap from '@plone/volto/components/manage/MaybeWrap/MaybeWrap';
import cx from 'classnames';
import type { BlockViewProps } from '@plone/types';
import type { ComponentType, ReactNode } from 'react';

const LegacyWrapper = (props: BlockViewProps & { children: ReactNode }) => {
  return (
    <div
      className={cx('block', props?.data?.['@type'], props.className)}
      style={props.style}
    >
      {props.children}
    </div>
  );
};

type BlockWrapperProps = BlockViewProps & {
  children: ReactNode;
  ExtraWrapper?: ComponentType<any>;
};

const BlockWrapper = (props: BlockWrapperProps) => {
  const isBlockModelv3 =
    props.blocksConfig?.[props.data?.['@type']]?.blockModel === 3;

  const ExtraWrapper = props.ExtraWrapper;
  return (
    <MaybeWrap {...props} condition={!isBlockModelv3} as={LegacyWrapper}>
      {ExtraWrapper ? (
        <MaybeWrap {...props} condition={!isBlockModelv3} as={ExtraWrapper}>
          {props.children}
        </MaybeWrap>
      ) : (
        props.children
      )}
    </MaybeWrap>
  );
};

export default BlockWrapper;
