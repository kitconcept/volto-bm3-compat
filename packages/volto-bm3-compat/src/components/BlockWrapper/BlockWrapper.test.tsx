import React from 'react';
import { render } from '@testing-library/react';

import BlockWrapper from './BlockWrapper';

describe('<BlockWrapper />', () => {
  it('renders the component wrapped in the specified component', () => {
    const blocksConfig = {
      button: {
        id: 'button',
        title: 'Custom block',
        blockModel: 3,
      },
    };
    const data = {
      '@type': 'button',
    };
    const BlockView = (props) => <div>The block view</div>;
    const props = {
      blocksConfig,
      className: 'extra-classname flex items-center bg-quanta-sapphire',
      data,
      style: { '--blockWidth': 'var(--container-default-width)' },
    };
    const { container } = render(
      <BlockWrapper {...props}>
        <BlockView />
      </BlockWrapper>,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders the component not wrapped (condition false)', () => {
    const blocksConfig = {
      button: {
        id: 'button',
        title: 'Custom block',
      },
    };
    const data = {
      '@type': 'button',
    };
    const BlockView = (props) => <div>The block view</div>;
    const props = {
      blocksConfig,
      className: 'extra-classname flex items-center bg-quanta-sapphire',
      data,
      style: { '--blockWidth': 'var(--container-default-width)' },
    };
    const { container } = render(
      <BlockWrapper {...props}>
        <BlockView />
      </BlockWrapper>,
    );

    expect(container).toMatchSnapshot();
  });
});
