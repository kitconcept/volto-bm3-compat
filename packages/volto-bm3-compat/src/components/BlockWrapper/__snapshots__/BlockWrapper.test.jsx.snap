// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<BlockWrapper /> > renders the component not wrapped (condition false) 1`] = `
<div>
  <div
    class="block button extra-classname flex items-center bg-quanta-sapphire"
    style="--blockWidth: var(--container-default-width);"
  >
    <div>
      The block view
    </div>
  </div>
</div>
`;

exports[`<BlockWrapper /> > renders the component wrapped in the specified component 1`] = `
<div>
  <div>
    The block view
  </div>
</div>
`;
